web-marketplace/src/pages/BuyCredits/BuyCredits.Form.tsx

```diff
@@ -311,6 +324,20 @@ export const BuyCreditsForm = ({
             });
           }
 
+          if (name && activeAccount?.id && !activeAccount?.name) {
+            updateAccountById({
```

Comment:
best to wait for this to succeed before refreshing profile data
```suggestion
            await updateAccountById({
```